diff --git a/hw/arm/versatilepb.c b/hw/arm/versatilepb.c
index dea5fc7..8150f43 100644
--- a/hw/arm/versatilepb.c
+++ b/hw/arm/versatilepb.c
@@ -277,9 +277,9 @@ static void versatile_init(MachineState *machine, int board_id)
 
     /* The versatile/PB actually has a modified Color LCD controller
        that includes hardware cursor support from the PL111.  */
-    dev = sysbus_create_simple("pl110_versatile", 0x10120000, pic[16]);
+    /* dev = sysbus_create_simple("pl110_versatile", 0x10120000, pic[16]); */
     /* Wire up the mux control signals from the SYS_CLCD register */
-    qdev_connect_gpio_out(sysctl, 0, qdev_get_gpio_in(dev, 0));
+    /* qdev_connect_gpio_out(sysctl, 0, qdev_get_gpio_in(dev, 0)); */
 
     sysbus_create_varargs("pl181", 0x10005000, sic[22], sic[1], NULL);
     sysbus_create_varargs("pl181", 0x1000b000, sic[23], sic[2], NULL);
